### PR TITLE
Revert "Bratseth/instances in deployment xml rebased take 2"

### PR DESCRIPTION
**Pull request by bratseth**
Reverts vespa-engine/vespa#10912

### CODE REVIEW

**Comment by bratseth** [comment]
Still some CD failures.

**Comment by jonmv** [comment]
Yup.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 1]
```diff
@@ -627,8 +628,8 @@ private ActivateResult unexpectedDeployment(ApplicationId application, ZoneId zo
     private LockedApplication withoutDeletedDeployments(LockedApplication application, InstanceName instance) {
         DeploymentSpec deploymentSpec = application.get().deploymentSpec();
         List<Deployment> deploymentsToRemove = application.get().require(instance).productionDeployments().values().stream()
-                                                          .filter(deployment -> ! deploymentSpec.requireInstance(instance).includes(deployment.zone().environment(),
-                                                                                                                                    Optional.of(deployment.zone().region())))
+                                                          .filter(deployment -> ! deploymentSpec.includes(deployment.zone().environment(),
```
[the remark this answers is not in the document]
Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Should only be needed for the one who deploys to production I think?  I sent a PR to remove that test.

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionPreparer.java [hunk 2]
```diff
@@ -122,8 +121,7 @@ public ConfigChangeActions prepare(SessionContext context, DeployLogger logger,
                 preparation.writeTlsZK();
                 var globalServiceId = context.getApplicationPackage().getDeployment()
                                              .map(DeploymentSpec::fromXml)
-                                             .map(spec -> spec.requireInstance(context.getApplicationPackage().getApplicationId().instance()))
-                                             .flatMap(DeploymentInstanceSpec::globalServiceId);
+                                             .flatMap(DeploymentSpec::globalServiceId);
```
[the remark this answers is not in the document]
Comment (jonmv) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionPreparer.java]: Haha. Only check the front page on mobile, not email :p

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Yes, when we orchestrate things, we'll never deploy illegally like this.

**Pull request merged**